Apps/APN-UIKit/APN UIKit/Util/DeepLinksHandlerUtil.swift

```diff
@@ -19,18 +19,32 @@ extension AppDeepLinksHandlerUtil {
     // Call this function if you have confirmed the deep link is a `deeplink` deep link. This function assumes you
     // have confirmed that.
     private func handleDeepLinkAction(_ url: URL) -> Bool {
-        let userInfo = ["linkType": "Deep link", "link": url.host ?? ""]
-        if let _ = storage.userEmailId, let _ = storage.userName {
+        if let host = url.host?.split(separator: "&"), host.first == "settings" {
+            var userInfo: [String: String] = [String: String]()
+            if host.count >= 3 {
+                // Site ID
+                let siteIdInfo = host[1].split(separator: "=")
+                if siteIdInfo.first == "site_id" {
+                    userInfo["site_id"] = String(siteIdInfo[1])
+                }
+                // API Key
+                let apiKeyInfo = host[2].split(separator: "=")
+                if apiKeyInfo.first == "api_key" {
+                    userInfo["api_key"] = String(apiKeyInfo[1])
+                }
+            }
+            if let _ = storage.userEmailId {
+                NotificationCenter.default
+                    .post(
+                        name: Notification.Name("showSettingsScreenOnDashboard"),
+                        object: nil,
+                        userInfo: userInfo
+                    )
+                return true
+            }
```

Comment:
I dont't think we should hardcode the positioning optional parameters. 

maybe, something like this,

```
if let urlComponents = URLComponents(string: url.absoluteString), let host = urlComponents.host, host == "settings" {
    var userInfo: [String: String] = [:]
    
    urlComponents.queryItems?.forEach { queryItem in
        if queryItem.name == "site_id" || queryItem.name == "api_key" {
            userInfo[queryItem.name] = queryItem.value
        }
    }
    
    if storage.userEmailId != nil {
        NotificationCenter.default.post(name: Notification.Name("showSettingsScreenOnDashboard"), object: nil, userInfo: userInfo)
        return true
    }
}
```

Reply:
Thanks for the suggestion, Shahroz. The deeplink that we receive is `apn-uikit://settings&site_id=xxx&api_key=yyy` of which `apn-uikit` is scheme and `settings&site_id=xxx&api_key=yyy` is the host. On trying to fetch queryItems it returns nil since the deep link format is not query friendly. If the decided deeplink had been `apn-uikit://?settings&site_id=xxx&api_key=yyy` i.e in format of a query string then the above suggestion would have worked.

Reply:
I agree with Shahroz with the suggestion. 

The deep link URL is not `apn-uikit://settings&site_id=xxx&api_key=yyy` (not a valid URL) but the URL that our sample apps receive is: `apn-uikit://settings?site_id=xxx&api_key=yyy`

Therefore, I believe his code sample would work.